x-pack/legacy/plugins/apm/server/lib/settings/cm/create_cm_index.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Server } from 'hapi';
+import Boom from 'boom';
+
+export async function createCmIndex(server: Server) {
+  const index = server.config().get<string>('apm_oss.cmIndex');
+  const { callWithInternalUser } = server.plugins.elasticsearch.getCluster(
+    'admin'
+  );
+  const indexExists = await callWithInternalUser('indices.exists', { index });
+  if (!indexExists) {
+    const result = await callWithInternalUser('indices.create', {
+      index,
+      body: {
+        mappings: {
+          properties: {
+            '@timestamp': {
+              type: 'date'
+            },
```

Comment:
Good point. The original issue describes a documents with this example:
```
{
  "_id": "s4uMZJqvfrNo7sa",
  "settings": {
    "sample_rate": 0.5
  },
  "@timestamp": "2019-04-12T14:00:17.700Z",
  "service": {
    "name": "opbeans-node",
    "environment": "prod"
  }
}
```
So based on this i used type `date` for `@timestamp`, but then the existing code that i started with was setting and reading from `timestamp.us`, so by writing to that field, it automatically updates the mapping to make the new field valid.

```
"timestamp" : {
    "properties" : {
    "us" : {
        "type" : "long"
    }
    }
}
```

I'm not sure the best way to proceed here. it looks like we use both `@timestamp` (date) and `timestamp.us` (long) in the apm indices. But we don't need both here. We also don't really need the microseconds granularity since only the UI write and reads from, so it's always `Date.now()+1000` anyways and is never set to anything more specific.

My intuition tells me to go with `@timestamp` (date) with epoch time in milliseconds, since that's what the UI needs.

Reply:
Yeah, I'm fine with either as long as we stick to one.